privacyidea/lib/tokens/daypasswordtoken.py

```diff
@@ -0,0 +1,516 @@
+from __future__ import division
+import logging
+import time
+import datetime
+from privacyidea.lib.tokens.HMAC import HmacOtp
+from privacyidea.lib.config import get_from_config
+from privacyidea.lib.log import log_with
+from privacyidea.lib.tokenclass import TokenClass
+from privacyidea.lib.tokens.hotptoken import HotpTokenClass
+from privacyidea.lib.decorators import check_token_locked
+from privacyidea.lib.policy import ACTION, SCOPE, GROUP, Match
+from privacyidea.lib.tokens.totptoken import TotpTokenClass
+from privacyidea.lib.utils import determine_logged_in_userparams, parse_time_sec_int
+from privacyidea.lib import _
+
+optional = True
+required = False
+
+log = logging.getLogger(__name__)
+
+class DayPasswordTokenClass(TotpTokenClass):
+
+    previous_otp_offset = 0
+
+    desc_timestep = _('Specify the time step of the DayPassword token.')
+
+    @log_with(log)
+    def __init__(self, db_token):
+        """
+        Create a new day password token object from a DB Token object
+
+        :param db_token: instance of the orm db object
+        :type db_token:  orm object
+        """
+        TokenClass.__init__(self, db_token)
+        self.set_type("daypassword")
+        self.hKeyRequired = True
+
+    @staticmethod
+    def get_class_type():
+        """
+        return the token type shortname
+
+        :return: 'daypassword'
+        :rtype: string
+        """
+        return "daypassword"
+
+    @staticmethod
+    def get_class_prefix():
+        """
+        Return the prefix, that is used as a prefix for the serial numbers.
+        :return: DayPasswordToken
```

Comment:
please add a new-line between the text and the parameter description

Reply:
https://github.com/privacyidea/privacyidea/pull/3602/commits/1d33e4a5414b7c3962159450ba40a6e9f9e29488